common/git.go

```diff
@@ -494,3 +503,44 @@ func CloneAndCheckoutBranch(testing *testing.T, repoURL string, branch string, c
 
 	return nil
 }
+
+// ChangesToBePush determines if there are any changes to push to the remote repository.
+// Returns a boolean indicating if there are changes and a slice of filenames that have changes.
+func ChangesToBePush(testing *testing.T, repoDir string) (bool, []string, error) {
+	return changesToBePush(testing, repoDir, &realGitOps{})
+}
+
+func changesToBePush(testing *testing.T, repoDir string, git gitOps) (bool, []string, error) {
+	output, err := git.executeCommand(repoDir, "git", "status", "--porcelain")
+	if err != nil {
+		logger.Log(testing, "Failed to determine if there are changes to push:", err)
+		return false, nil, err
+	}
+
+	// If no output, return false and empty slice
+	if len(output) == 0 {
```

Comment:
If output is nil this will panic. I know its not expected, but I've been trying to get better at checking for this just to make sure. One extra check for nil might save it.